src/mpid/ch4/netmod/ofi/ofi_win.h

```diff
@@ -56,6 +53,24 @@ static inline int MPIDI_OFI_win_progress_fence_impl(MPIR_Win * win, bool do_free
         }
     }
 
+    /* We need to poll the CQ to process any completion events corresponding
+     * to the progress made by reading the counter.
+     * Since we don't know what the value of the fi_cntr was before reading
+     * it, we might have made progress by reading it. Hence, we need to poll
+     * the CQ at least and then for `max_cq_reads` times, if needed */
+    itercount = 0;
+    max_cq_reads = (tcount - firstcount) / MPIDI_OFI_NUM_CQ_ENTRIES;
```

Comment:
Should this be `(tcount - firstcount + MPIDI_OFI_NUM_CQ_ENTRIES - 1) / MPIDI_OFI_NUM_CQ_ENTRIES`?

If `tcount - firstcount = 10` and `MPIDI_OFI_NUM_CQ_ENTRIES = 8`, we may want to call OFI progress twice, not once.

Reply:
Yes, that's right. I think it should be `(tcount - firstcount) / MPIDI_OFI_NUM_CQ_ENTRIES + 1`?

Reply:
Technically that would return incorrect value when `(tcount - firstcount)` is multiple of `MPIDI_OFI_NUM_CQ_ENTRIES`.

Reply:
You are absolutely right :)